### ISSUE TITLE
[VERSION TRACK] v3.4.1 Known Issues & Bugs

### ISSUE THREAD

**Issue by jtsage** [FSGModding/FSG_Mod_Assistant#210]
# Known Bugs - v3.4.1

This issue is an umbrella issue to keep track of bugs in the current version.  You can reply directly here, but opening a new issue would make life easier for us.

## Issue List (included in 3.4.1)
- [x] (3.4.1) Misspelling of 2 new fill types from straw harvest (or variants) 954c80174d3c9cb6687f6cef8295b2e9f684d079
- [x] (3.4.1) Dumb logic typo for base game placeables (they all have the "parent file" warning background) 72f1325edd55c42336d4d76d66243c5d7f03893d
- [x] (3.4.1) Game running indicator not hiding as expected. 0d06dc5c9632db7530cb49c3fae3c17e032c78bf
- [x] (3.4.1) Chart Units missing on mod detail pages ab26a44e1d71990bc363908b1804d71824f99103
- [x] (3.4.1) Unusable mods not visually different in mod list f1142e69a011449b199a9b9aee13eb48423c2097
- [x] (3.4.1) Documentation was very, very outdated. ce3383290acb48c3e9d996e950d91bc055134698, d61773d9a1120ef81e03d33407bb26ae15d9aba5, 4d5daab5a5d744c5d2cc97868bee0685ba851fa6, 855b55dcd321f0382a712d87d158828a47efdb30

## Issue List (included in 3.5.0-RC1)
- [x] (3.4.91) Add logic trap to remove `/x86/` from game path on base game browser. #211 c663d5296a0b477d7c33c90a98dfc4f5261daae3
- [x] (3.4.91) Download display was wonky, mis-placed text, and the speed calculation failed at 1,000Mb. Also allow https:// in server URL's. 91b1d5f42f1245a334b17bdaca9cde4cfaaa42e4
- [x] (3.4.91) Visual indication needed on mini-mode when game is running. 7850107a44dabd7b8a10d789e607ab16de988957
- [x] (3.4.91) Add option to drag-and-drop a CSV collection export to the main window to compare it to current collections.  Uses the same visual interface as the savegame compare tool. #208 7850107a44dabd7b8a10d789e607ab16de988957, 278fcc2d03cac68144870688a9397eeb7c4ba5c1 
- [x] (3.4.91) Deep link of attacher joints in mod detail display broken 4ecd658d32931f898975fb8aeaa70921fe04f91a
- [x] (3.4.91) Button list for collections was getting long, prefer icon for savegame checker. ab1a8f3e88277577bd689fee7ca8e015040da1eb
- [x] (3.4.91) Preferences window is clunky - #212 0a5fbcc6b713b4ca26f2ff8a266f9a7dd1728be6, f02cd6e1d44ef9d1c8834582f585100e76ba1de8
- [x] (3.4.91) Collection Management in separate window is clunky - #213  0a5c3387224cc42419fb94d38888ac7bcf808eeb
- [x] (3.4.91) Code duplication in preload files, some can probably be shared. fd045036185a7d075f3f0537e9f2981df1ec2a2c
- [x] (3.4.91) Disabling older FS does not disable XML file checking #214 0c18f1312cbab5136279ada40bfe5be842ceabd4
- [x] (3.4.91) Game Polling (status bubble) should be optional. 1b630fa0de123e699df4d4b71c67af1ba5ca6a4f
- [x] (3.4.91) Collection details/notes page should visually resemble settings page. 193cc6dc3f392c46c5f1ae40595fb9094a1ca1b6

## Issue List (included in 3.5.0-RC2)
- [x] Missed some log spam for game polling 15760ebd63aff25dbb97f9d9eb9693433eb8d2da
- [x] Checkbox on preferences / collections details is unclear, use a custom option [diff](https://github.com/FSGModding/FSG_Mod_Assistant/commit/3bf6245777fff8e255059c7e462946bf16bf5fee#diff-455e0d9c324b160ae695771a8fb00aa98d4abb0a450224d28092b66e4f30d831) 3bf6245777fff8e255059c7e462946bf16bf5fee
- [x] Visual overlap on gamelog screen #215 7eff21642c5f95c6833f75b7af92419e69c4c46a
- [x] Right click to copy/move/delete all selected would be nice ae205dcf5126f6f96e26ee51fedc62a7a19e463c
- [x] Not happy with the spacing / alignment of side buttons since change to configurable font-size f7523ac2a2d818bd3d679142ca8af01bd68115a6
- [x] Right click on a savegame had an icon code, not text.  fix in l10n files. f23e09e8d536b62c1d54385ef80b23cccf7dd671

## Issue List (included in 3.5.0-RC4)
- [x] Add GitHub Cat to button 4ba2740fd0d7a2e4c1e3df7fa74ae7d3844aba3e
- [x] Add Icons to right click menus 228c6d607961159d011cbeddafa0805ff885c91c
- [x] Hide Crop Calender JSON display when not applicable, turn into a copy-to-clipboard button otherwise. fa6a3bb3ef9293514149cf6aefe9208600b183c5, d5a7a70a48f0ed9cd5c145d683136c30bb89a1ec
- [x] Erase search box icon is off-center when font-size is not default. b3be9c1e81db9083cbdc7af4f618350a9cedfa0f
- [x] Open log file in explorer button not working a0a7d0e9cc037cd73cce231be6fe83f036bd71a7
- [x] FSG Bot integration (server status) would be pretty spiffy. e2dee3aaa0c52522b9ce443a7df9cf5c38dc1e16, 39eca690958f57e5b275bf45d1f3f1ec3cc319d8 [howto](https://fsgmodding.github.io/FSG_Mod_Assistant/fsgbot.html)
- [x] Need light variant (black) context menu icons as well. 0bcca1558e46daf43d86e3b514f7f457d467f15c, 2e4c05db35ff50a5fb569e41fe78bdef451520bd
- [x] Support for "Through the Years" mod - display the vehicle year in the details, if it is known. 29262f26a9ebe287d41d2f51c548dcca3e04836c

## Issue List -- Included in 3.5.0-RC6, most in #218 
- [x] Context menus follow the system theme, not the MA theme - this is probably wrong. 44823e21334f08ceb96d214d08990b0e85bdaf5b
- [x] A first-time setup wizard sure would be cool
- [x] The move/copy/delete/import windows shouldn't be separate windows, move them where they belong
- [x] Updates/changes got unwieldy again, refactor main file for readability and organization
- [x] Add season temperature detection for maps
- [x] Many visual changes to use inset-box model from preferences pane
- [x] CTRL+Space shortcut works in mini window now
- [x] drag-and-drop should not fire when file operation or preferences change is in progress
- [x] New tag filtering system using "advanced" radio buttons - I think it makes more sense as to what is happening
- [x] Add a "compare all" button to base game sub categories (e.g. "Large Tractors")
- [x] Try and plug a memory leak in the icon converter / generator
- [x] Share savegame compare results with the main window for as long as the savegame compare window remains open.

## Issue List - Included in 3.5.0-RC7
- [x] Fix basegame graphs again
- [x] Catch modlook fail from missing folder/file
- [x] Make depends / required by submenus useful
- [x] automate HTML testing
- [x] Add icons to tray menu
- [x] Show dropdown when collection has multiple maps
- [x] Detect malicious mods (LUA that nukes your mods, saves and settings)

Special Note: there is no way I didn't break a few things on this RC, at least 1 or 2 more are going to be forthcoming before 3.5.0-RELEASE.

## Progress
- [x] BugFix 3.4.1 Released : [v3.4.1](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.1)
- [x] 3.5.0-RC1 Released : [v3.5.0-RC1](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.91)
- [x] 3.5.0-RC2 Released : [v3.5.0-RC2](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.92)
- [x] 3.5.0-RC3 Released : [v3.5.0-RC3](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.93)
- [x] 3.5.0-RC4 Released : [v3.5.0-RC4](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.94)
- [x] 3.5.0-RC5 Released : [v3.5.0-RC5](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.95)
- [x] 3.5.0-RC6 Released : [v3.5.0-RC6](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.96)
- [x] 3.5.0-RC7 Released : [v3.5.0-RC7](https://github.com/FSGModding/FSG_Mod_Assistant/releases/tag/v3.4.97)

**Comment by vitalii2011** [comment]
This is probably overkill for non-cards:
![Image 001](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/f5e14990-83b3-4447-9a79-5cfa8ff0ebf9)

**Comment by jtsage** [comment]
Oh, good catch - yeah, it should hide it if it doesn't have it.  honestly, it should probably always be hidden behind a click-gate, it's pretty specialized as to who needs it :)

**Comment by vitalii2011** [comment]
P.S.
In addition to maps with a “calendar”, as you know ) , there are mods like Crop Calendar Manager https://www.farming-simulator.com/mod.php?lang=en&country=us&mod_id=275437

**Comment by jtsage** [comment]
> P. S. In addition to maps with a “calendar”, as you know ) , there are mods like Crop Calendar Manager https://www.farming-simulator.com/mod.php?lang=en&country=us&mod_id=275437

Haha.  Yes indeed.  I actually spoke to Jos when he released this to nicely let him know I wasn't going to directly support it in MA :)  It's a very cool mod though.

**Comment by jtsage** [comment]
> Probably, there is space left here for command icons like in Windows Explorer?

I don't know.  But that looks rather nice with the icons, and I will find out for sure.

**Comment by jtsage** [comment]
![context-collection](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/30895c86-3774-4f67-9f23-9969d6b69a2a)
![context-cutcopypaste](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/82c04933-1263-4b09-b913-367bc0a37074)
![context-find-all](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/13050d15-2c9c-4d93-99ff-1b476610c9a6)
![context-mod-depend](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/85de7b04-8129-488a-904c-7f928b79891b)
![context-mod-log](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/f96c3989-f579-4c68-a528-20ede6e20b17)
![context-mod-main](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/6549f855-5e47-4c69-9ee1-67bc73a618d0)
![context-mod-save](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/bbddc4a5-fc62-47d2-b1c0-ca722f471e72)

**Comment by vitalii2011** [comment]
I don't understand why the "heart" is there. They are not necessarily looking for what they “like”. 
![300903461-85de7b04-8129-488a-904c-7f928b79891b](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/d3f7182f-d9a6-4eec-bac1-55b6b07f98e5)
But this, of course, is just my opinion. )

As always, you are doing much better and on a larger scale than expected.

**Comment by jtsage** [comment]
It's to match the button on the side - I think I picked it because it was the only option that wasn't just the spy glass.  I'm open to other ideas as to what could mean "modhub" - I came up pretty empty.

**Comment by vitalii2011** [comment]
Just a no-frills magnifying glass.
And it doesn’t matter that the same will happen for another “Find”. 
Search is search. )

**Comment by vitalii2011** [comment]
![Screenshot 2024-01-30 at 21-16-50 ModHub Farming Simulator](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/88f83e71-e86a-48a0-9350-47cae647a68c)

**Comment by vitalii2011** [comment]
By the way – “binoculars”?

**Comment by vitalii2011** [comment]
"Add Icons to right click menu" is not available in 3.4.93.

**Comment by vitalii2011** [comment]
Hide Crop Calender JSON display when not applicable,
![Image 001](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/7bae92c7-693f-4fe4-bc49-c16f0edc1fcf)

**Comment by jtsage** [comment]
> "Add Icons to right click menu" is not available in 3.4.93.

huh.  that's odd.  I have it, on the work computer I just installed on (not my work machine)

![2024-01-31 09_56_59-FSG Mod Assistant](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/9203237e-9bfa-41b5-b6fa-3f84104a50e6)

So, something is odd here.  I'll try and reproduce.

**Comment by jtsage** [comment]
> "Hide Crop Calendar JSON display when not applicable" - in the same place, but in the form of a button: ![Image 001](https://private-user-images.githubusercontent.com/41552153/301050014-7bae92c7-693f-4fe4-bc49-c16f0edc1fcf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MTMyOTIsIm5iZiI6MTcwNjcxMjk5MiwicGF0aCI6Ii80MTU1MjE1My8zMDEwNTAwMTQtN2JhZTkyYzctNjkzZi00ZmU0LWJjNDktYzE2ZjBlZGMxZmNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE0NTYzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzYzAxY2E4NmYyMGY3MWFkNTdmNzc3OWRhODkwZDdmNzdiMjFiNjQ0NDQyZmNhODhlNTZhMjkzNDgwNzZmMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UHmQdnOlyQ0Efz5rJT2M02PSMfTChhDI0x20b27Ut-o)

Lol.  this one is on me, I hid the header, but the button isn't a child of the header.  my bad.

**Comment by vitalii2011** [comment]
> huh. that's odd. I have it, on the work computer I just installed on (not my work machine)
I only looked in the Search field.
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/5113a99f-70af-436e-9671-633d0508afda)
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/2da82d34-b399-4987-821b-0759c20bf74f)
![0012](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/50c6279b-dc53-4490-a698-d052889d04bb)

**Comment by jtsage** [comment]
LOL!  ok.  they are in fact there, probably.  But they are white.  I didn't do any sort of theme checking on it. I'll get one that :)

**Comment by vitalii2011** [comment]
> But they are white. 

That's what I assumed. )

**Comment by vitalii2011** [comment]
Is it possible to navigate by clicks or the ability to copy the title?
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/e258ceef-09a7-416b-833b-f8554c8581ac)
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/593c776f-5469-4268-8b25-728c8d3c12fe)

**Comment by jtsage** [comment]
Yeah, I can make those do something - tell me more on your idea - open up the details page for the clicked dependency / requirement?  (I have to check if those lists are populated only with existing mods - it might be everything, even if the "requirement" is not met)

**Comment by vitalii2011** [comment]
- Indeed, some scripts are not on ModHub. Then just copy the name. It seemed like: scroll the list to the mod whose name you clicked on.
- There is an option: search for the missing mod on the Internet, but for me, for example, this is not suitable - Google search in my Republic is blocked by providers. Then you will need a choice of search engines, and this is unnecessary.

**Comment by jtsage** [comment]
that makes sense.  ok.  I'll do a "scroll to the mod" if it exists.  If not, I'll have it populate the system clipboard with the name of the mod, along with a little bubble popup that says it was not found, and it's available to paste into whatever of your choice.

**Comment by vitalii2011** [comment]
It would probably be more convenient if these category labels
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/c00cd784-ac75-4e7c-9684-e8bf680611b9)
 were displayed in the main list:
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/d08b234c-04f6-426d-8809-b2e4846db389)

**Comment by jtsage** [comment]
Oh.  yeah, I see what you mean.  Ok.  I'm gonna need to think about how to achieve that - the results in the savegame compare window are pretty transient, but I should be able to populate them back to the main window.

While you are around - do you have any thoughts on the copy/move/delete operation being a separate window? Would it better serve as a overlay to the main interface?

I had hoped to get 3.5.0 released today - however, I've gotten involved in some style changes in some windows, and adding an initial setup wizard, so that is not going to happen.  I hope to have a new RC build late tonight though.  Fair warning, there are more than a few new translation strings for the initial setup wizard that I will add as soon as I can take decent screenshots :)

**Comment by vitalii2011** [comment]
> While you are around - do you have any thoughts on the copy/move/delete operation being a separate window? Would it better serve as a overlay to the main interface?

I think this is not necessary. For what?
If I get you right.

**Comment by jtsage** [comment]
These windows.

![2024-02-04 10_48_06-FSG Mod Assistant](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/5323f56b-f19a-4524-b7b8-6300f12892c4)

My thoughts **for** making it a overlay of the window:
 - less extra windows opening. (both easier and slightly harder from a programming perspective)
 - I'm using an "always on top" flag on those - but I don't have the option to make it always on top of only other Mod Assistant windows - I find that frustrating sometimes.

My thoughts **against** making it an overlay:
- The way it is now, you can still scroll in the main window if you move it out of the way if you have a reason to need to do that.

**Comment by vitalii2011** [comment]
I have never moderated a group game. Perhaps such an option is useful/convenient for such moderators, but I have no idea about it.

**Comment by jtsage** [comment]
> It would probably be more convenient if these category labels

This one was kinda hard.  But it is tied to the open save game compare window, when you close that, the tags disappear again.  They also *only* appear on the collection you have compared to (limitation in how I'm doing the comparison, when these tags go back to the main window it's too late to figure out if they were not present in the compared collection but do exist in the new one you are viewing)

![2024-02-08 12_48_45-](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/fcdce7e1-82cc-4361-a8fd-12279aca88ea)

as per usual, I'm still messing with exactly how the tags are displayed.  Alas, it is also too late to add them to the filter buttons next to the search box.

**Comment by vitalii2011** [comment]
Sorry, I forgot ( , which can be done like this:

https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/dbfe0d6a-2d9e-4661-b7bd-c6f13ed93d63

**Comment by vitalii2011** [comment]
It might make sense to add a "Selected" button:
![bandicam 2024-02-08 21-20-40-322 mp4_20240208_212128 875](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/ef51562e-81f4-4c17-906a-ad48c111bec9)

**Comment by jtsage** [comment]
aye.  it's been a space problem - but part of the update is a rework to how the two filter buttons appear, so there is likely room there now

**Comment by vitalii2011** [comment]
Video capturing all the necessary windows:

https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/0e7f8d51-9efb-4947-9829-e2e7e22d4e5d

**Comment by jtsage** [comment]
sweet, thank you - I had thought that was what you meant - added to my working copy.  I have a bunch to do as far as updating screenshots and documentation, but I hope to have the updated translations to crowdin and a test build sometime tonight (UTC-05:00)

**Comment by vitalii2011** [comment]
Video capturing all the necessary windows:

https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/1910d451-1ee8-402c-a092-93608d553ca9

**Comment by vitalii2011** [comment]
3.4.96 - strange/not always the close button “Select game version” works: when the window is minimized, with several clicks it expands the window to the full screen and only then works:

https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/4759dfd0-8f81-44b9-b9d1-48b2795a1432

- Here's another place where there are no icons for menu items ) :
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/8d86ae44-530a-49ac-81fd-ef44be0f8b9f)
- I didn’t find anything “good” anymore. ))) 
- Liked the maps icon:
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/fe1ad9f9-5054-4317-b37a-7bc7aa50eef0)

**Comment by jtsage** [comment]
I really should already be sleeping - but thanks, I'll take a look in the morning closer.

As to that close button - on closer inspection, it looks like it only works if you click exactly on the white X (not the background doesn't count).  I'll adjust to the more standard button style.

Good catch on the context menu icons, I forgot that one was there.

As to the map icons - yes.  I'm still playing with a way to reasonably display more than one - as of right now, it will show a map icon if there is one, and only one map in the folder.  Maybe a hover/dropdown combo or something if it has more than one would look nice

**Comment by vitalii2011** [comment]
- This huge planet: when someone just wakes up, someone else's day will already be over. 
- Maybe it’s possible/easier to add a mod to “Selected” when click it?
![0012](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/a5a6159f-2fe1-4ac0-b670-4ef418b77f43)

**Comment by jtsage** [comment]
ok, I think this is pretty clever...

![2024-02-09 11_37_01-modAssist_func_lib js - FSG_Mod_Assistant (Workspace) - Visual Studio Code](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/02793f77-f8ba-4cf9-bbe9-b87c03249813)

Thumbs Up - you have it, in the same collection, all is well - clicking on it will do what you described - select it in the main window, with the search field filled to just that mod.

Thumbs Down - you don't have it anywhere, the mod short name (everything before the `.zip`) will be copied to your keyboard

Spy Glass - you have it, but it's in the wrong collection - clicking it will bring up the "Find All" window with that mod pre-selected so you can grab it from the collection of your choice and copy it around if you want.

**Comment by jtsage** [comment]
![2024-02-09 12_11_46-](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/cfb12d17-036d-433a-844d-f3e026f26656)

**Comment by vitalii2011** [comment]
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/434e026b-36b4-4e62-8495-267ea2fd3c99)

**Comment by vitalii2011** [comment]
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/7af3cff6-7708-4371-9ba0-8834f9c1bcab)

**Comment by jtsage** [comment]
yeah, you're probably right - I wasn't sure I could mess with those after I created them, but in getting the theme right I had to figure out how to do just that anyway

**Comment by vitalii2011** [comment]
If you can change the background behind the image with the tag name:
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/b5042c17-597a-4f5a-bde5-4800554b283e)

**Comment by jtsage** [comment]
![2024-02-11 13_08_27-FSG Mod Assistant](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/178668c3-88a7-445c-8dd7-9b87f6156728)

multiple-map dropdown for collections with more than 1

**Comment by jtsage** [comment]
> If you can change the background behind the image with the tag name: ![0010](https://private-user-images.githubusercontent.com/41552153/303796634-b5042c17-597a-4f5a-bde5-4800554b283e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc2NzUyNDEsIm5iZiI6MTcwNzY3NDk0MSwicGF0aCI6Ii80MTU1MjE1My8zMDM3OTY2MzQtYjUwNDJjMTctNTk3YS00ZjVhLWJkZTUtNDgwMDU1NGIyODNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjExVDE4MDkwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhYzk2YWZiNDE0OTU3NzU5NDQ0MWU5MDhiNDI4ZjlhNDc2NmNhNDAyYjkzNjE3MzE0YzU3Njk4OWQ0MmY2MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2TX-78nYJH1YQvwDMSXIplnjNYdvHvbyzmr4ZJ8kQQs)

I *can* change the background, yes - where I am having trouble is in using the standard red/yellow/green traffic light design to communicate what is really happening in this dropdown.

The buttons correlate pretty closely to the RFC standard definitions : https://datatracker.ietf.org/doc/html/rfc2119

The PIN is the mod MUST have that tag (more than one pin, it MUST have all of them)

The SLASHED EYE is the mod MUST NOT have that tag (more than one slashed eye, it MUST NOT have any of them)

The OPEN EYE is the mod MAY have that tag (it does not need to, it's provided as a visual opposite of the SLASHED EYE)

Finally, the PIN causes the OPEN EYE to be forcibly toggle "on"

I can see how red would correlate well to the SLASHED EYE - but the green state could be *either* the PIN or the OPEN EYE, and that's where my confusion comes in.

**Comment by vitalii2011** [comment]
> раскрывающийся список с несколькими картами для коллекций, содержащих более 1

- Great!
- Are these targets for a mini-game?
![303938284-178668c3-88a7-445c-8dd7-9b87f6156728](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/c1168758-aa50-42fa-a6b9-672cc778f2fc)

**Comment by jtsage** [comment]
heh.  Bot integration with the bot our community leader wrote - it's really a multi-player specific feature, but it's kinda cool

https://fsgmodding.github.io/FSG_Mod_Assistant/fsgbot.html

In the case you posted, that would be our 16 regular realistic servers along with the 2 "co-operative" ones.

**Comment by vitalii2011** [comment]
> I _can_ change the background, … my confusion comes in.

I'll think about it. But it’s hard, because I’m not used to it. )

**Comment by vitalii2011** [comment]
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/d35b5f81-1855-4a5a-afbb-f08e4abede20)
When working with this list, I feel like clicking on the picture with the tag name, but now it’s not a button.
And that would probably be more convenient.
But and now everything is fine.

**Comment by jtsage** [comment]
I could add a click there - maybe have it just cycle to the "next" setting? show->hide->limit to->repeat?

**Comment by vitalii2011** [comment]
I think it’s normal now, but you know better.
![0011](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/9753ce22-bf9e-41c7-904b-4657ebfb7290)

**Comment by vitalii2011** [comment]
This https://discord.com/channels/949619926463840327/1009879187429003377 is very frustrating.
![Image 001](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/73b93c64-6d92-4bae-9baa-f4f86ffbb0a7)
![ScreenShot_20240211150212-1](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/1415d677-731c-47e8-ad46-82de45008b58)

**Comment by jtsage** [comment]
working on it right now.  it'll be in the next update

**Comment by jtsage** [comment]
I just pushed the changes for the malware detector - there are probably other false positives out there, but I did check it against the entire 'gameplay' category on modhub, along with the popular script mods that I use.  Please use issue #220 to report any others.  If you want a safe file to test with, see [here](https://github.com/FSGModding/FSG_Mod_Assistant/blob/main/test/tests/mods/EXAMPLE_Malicious_Code.zip) for my test mod, it has the offending code in a LUA file that is neither loaded nor able to actually run.

![2024-02-11 19_11_42-Possible Malware Detected](https://github.com/FSGModding/FSG_Mod_Assistant/assets/605986/0134c8c6-6397-4e34-b9c1-94d6994513c4)

**Comment by vitalii2011** [comment]
Users may have problems with 3.5.0-RC7!

**Comment by vitalii2011** [comment]
When "I do not trust it, delete it right now" are archives with mods deleted not to the Recycle Bin/bypassing the Recycle Bin?

**Comment by jtsage** [comment]
> When "I do not trust it, delete it right now" are archives with mods deleted not to the Recycle Bin/bypassing the Recycle Bin?

that's correct - they are removed from disk directly, it bypasses the recycle bin.  In this case, I could send it to the recycle bin - keep in mind, this is detecting mods with funky scripting (there were a whole bunch released in the last few days on some sites) - it is not, nor is intended to be a system wide malware scan - there are plenty of options for that already.  This is only looking for mods that may maliciously alter your farm sim installation only.

**Comment by vitalii2011** [comment]
At the first launch, the color of the text did not change when hovering the cursor (the next time it changed): 

https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/396d0bfb-8acb-45d8-be9c-32069032be0e

 it would be better if the arrow became a “pointing finger” - not everyone will be able to see the change in text color and understand that when clicked, an action from the text will occur.

**Comment by jtsage** [comment]
Oh, and some additional information on the rest of the buttons - the first one you mentioned does indeed delete the mod.

The second "I want to investigate this" adds it to an ignore list by hash and folder hash - if you move or copy the file to another collection, the warning will re-display for that copy of the mod.  Also, that ignore flag is not saved, so the next time you start Mod Assitant, the warning will show again.

Finally the "I'm sure I trust this..." (last button) adds it to a _saved_ list in your preferences, by name only - it will not show the "malware" tag, and it will not ask you about that mod ever again.

**Comment by jtsage** [comment]
> At the first launch, the color of the text did not change when hovering the cursor (the next time it changed):
> 
> it would be better if the arrow became a “pointing finger” - not everyone will be able to see the change in text color and understand that when clicked, an action from the text will occur.

Got it - when I ran the RC build I noticed some odd behavior as well - it looked like the dialog was getting hung up in or under the changelog of the splash screen - I'll look at the timing of events on those.  And i'll look at changing the pointer type as well.

**Comment by vitalii2011** [comment]
For some reason, MA didn’t like FS19_GlobalCompany:
![Image 001](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/b321c03c-3fb2-49ec-8c56-53dbe1af187f)
And deleted (due to my carelessness) VehicleInspector 
[VehicleInspector.zip](https://github.com/FSGModding/FSG_Mod_Assistant/files/14235339/VehicleInspector.zip)

**Comment by jtsage** [comment]
whoops.  I should have checked vehicleinspector - I forget what his other massive overlay one was called, but I bet that does it too.  And for some reason it never occurred to me to run it against my copy of 19 mods - I bet there's a fair few in there too. Thanks, I'll get them added prior to the next release

**Comment by vitalii2011** [comment]
That's why there was a question about the reason for deleting bypassing the Recycle Bin. Someone may delete something that they edited for a long time and/or cannot restore.

**Comment by jtsage** [comment]
> That's why there was a question about the reason for deleting bypassing the Recycle Bin. Someone may delete something that they edited for a long time and/or cannot restore.

I agree - I'll change this call to send it to the recycle bin instead.

**Comment by vitalii2011** [comment]
It might be possible to make a list of detected/removed malicious files somewhere here:
![0010](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/091f3000-afc3-4163-8d40-d5c72a902069)
Because this: 
![Image 003](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/3eae0f37-8ed3-45b8-b693-0a1ff43331c8)
 may remain undeleted (I understood your explanation of why they are not there) without tags, etc. 
Pests exist not only on the Internet) For example, a child can download various cool cars with nasty things and mark them all as “acceptable”.

**Comment by jtsage** [comment]
yeah, I will add what you have "permanently" ignored in there, along with a button to clear it out.  I was considering having that list clear itself on the "Clear Cache" button as well

**Comment by vitalii2011** [comment]
> EDIT: if you want to alter the translated text on that button to reflect that, I will do the same.

No. For what?
If I get you right. )

**Comment by jtsage** [comment]
> > EDIT: if you want to alter the translated text on that button to reflect that, I will do the same.
> 
> No. For what? If I get you right. )

I'll probably change that button from

`I do not trust it, delete it right now`

to something like

`I do not trust it, move it to the recycle bin it right now`

**Comment by vitalii2011** [comment]
> whoops. I should have checked vehicleinspector - I forget what his other massive overlay one was called …

https://www.modhoster.com/community/user/happylooser#t2
![Screenshot 2024-02-12 at 08-47-01 Profile of HappyLooser](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/77d42c8b-5ea0-4d2b-9bcb-2de546526a04)

**Comment by jtsage** [comment]
oh, cool, thanks - I've not used any of his stuff in a while - but I know he spent a ton of time on library and utility functions, so they probably all use similar code to manage saved settings which is causing the false positive on this

**Comment by vitalii2011** [comment]
https://github.com/loki79uk/FS22_NoDelete/releases/tag/v0.0.0.1

**Comment by vitalii2011** [comment]
![Image 001](https://github.com/FSGModding/FSG_Mod_Assistant/assets/41552153/2acf1c7d-0d48-4d3d-a322-4da40df1c8a8)
```
        local g_env = getmetatable(_G).__index
        g_env.deleteFile = function(path)
            print("WARNING script attempted to delete file: " .. path)
        end

        local function createTextFile(filename, text)
            local fileId = createFile(getUserProfileAppPath() .. filename, FileAccess.WRITE)
            fileWrite(fileId, text)
            delete(fileId)
        end
        createTextFile("testFile.txt", "test content")
        
        local userProfileAppPath = getUserProfileAppPath()
        local files = Files.new(userProfileAppPath)
        for _, v in pairs(files.files) do
            if v.filename == "testFile.txt" then
                print("hahaha I am deleting your testFile...")
                g_env.deleteFile(userProfileAppPath .. v.filename)
            end
        end
```

**Comment by jtsage** [comment]
> https://github.com/loki79uk/FS22_NoDelete/releases/tag/v0.0.0.1

Aye, Loki and I were talking about this yesterday a bit - looks to be a useful bit of protection.

For what it's worth, this mod was actually the second entry into the global whitelist for MA's scan, as his test case there also triggers the warning :)
